packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -264,6 +264,19 @@ async function pruneExecutionData(this: WorkflowHooks): Promise<void> {
 	}
 }
 
+async function saveExecutionMetadata(
+	executionId: string,
+	executionMetadata: Record<string, string>,
+): Promise<void> {
+	for (const [key, value] of Object.entries(executionMetadata)) {
+		await Db.collections.ExecutionMetadata.save({
```

Comment:
Can we do this as a transaction or a Promise.all()? Otherwise we'll introduce multiple delays here.